pybloqs/htmlconv/chrome_headless.py

```diff
@@ -0,0 +1,68 @@
+from io import open
+from numbers import Number
+import os
+import tempfile
+
+from pybloqs.config import ID_PRECISION
+from pybloqs.htmlconv.html_converter import HTMLConverter, PORTRAIT, A4
+
+
+_NODE_SCRIPT_LOC = os.path.dirname(os.path.realpath(__file__)) + "/puppeteer.js"
+
+
+class ChromeHeadlessConverter(HTMLConverter):
+    """
+    Implementation of HTML to PDF conversion using headless chrome,
+    accessed via puppeteer node.js bindings.
+
+    See package.json at the root of this project for the prerequisite node.js
+    dependencies
+    """
+
+    def htmlconv(self, block, output_file, header_block=None, header_spacing=None, footer_block=None,
+                 footer_spacing=None, pdf_zoom=1, pdf_page_size=A4, orientation=PORTRAIT, **kwargs):
+        """
+        Use headless chrome to convert the supplied html content to a PDF file.
+
+        :param block: The block to render
+        :param output_file: File name for converter output.
+        :param header_block: A header block to add
+        :param header_spacing: The spacing for the header, either HTML lenght unit or number (interpreted as mm).
+        :param footer_block: A footer block to add
+        :param footer_spacing: The spacing for the footer, either HTML lenght unit or number (interpreted as mm).
+        :param pdf_zoom: The zooming to apply when rendering the page.
+        :param pdf_page_size: The page size to use when rendering the page to PDF.
+        :param orientation: Either html_converter.PORTRAIT or html_converter.LANDSCAPE
+        :param kwargs: Additional parameters. Not currently used in chrome_headless backend.
+        """
+        zoom = pdf_zoom * 0.75  # Produces roughly same zoom as wkhtmltopdf
```

Comment:
This doesn't feel right to me, i'd rather take the argument exactly as provided by the caller - appreciate it's trying to make it a drop in replacement for the wkhtmltopdf converter but it makes this inconsistent with creating an html, opening it in chrome and saving the pdf yourself.